### ISSUE TITLE
Holdout reflections with RectLight not showing in alpha channel

### ISSUE THREAD

**Issue by DerFluep** [prman-pixar/RenderManForBlender#713]
Hi,
i can't get Holdout reflections to work when just using RecLights and rendering to IT or using batch render.
The Alpha channel just contains the shadow part but not the reflections.

First image is the scene without a holdout, second with holdout and third the alpha channel with holdout.
![no_holdout](https://github.com/user-attachments/assets/60740c8c-1714-4ef2-8ce5-be6e00dad78e)
![holdout](https://github.com/user-attachments/assets/bed9a36a-8857-4048-8e57-77dc8eaf98ba)
![holdout_alpha](https://github.com/user-attachments/assets/6469c92b-cba4-412e-b2a1-376a883e7453)

when using a DomeLight or EnvDayLight everything works as expected and the reflections show up in the alpha channel.
Is this intentional or am i missing something?

With an EnvDayLight:
![holdout2](https://github.com/user-attachments/assets/21e16690-6faa-4e66-a698-ab5a106449ce)
![holdout_alpha2](https://github.com/user-attachments/assets/172807b7-5caa-49d4-93fd-3da72e0d001c)

but when rendering directly to blender it works again, i think because the alpha is directly baked into the beauty pass?

Here are two simple szenes 
[example.zip](https://github.com/user-attachments/files/17894665/example.zip)

**Comment by prman-pixar** [comment]
Hmm...I'm not sure what's going on here. I don't think this is a problem with the addon itself, as the same thing seems to happen in our Maya and Houdini plugins. 

It seems to work in our regression tests, so I'm still trying to figure out what our DCC plugins are doing incorrectly.

**Comment by prman-pixar** [comment]
@Corefrecs  

I'm still not sure why you're not seeing the reflection in the alpha, but I think if you want to composite this over a plate the best way to do this would be:

1. Change the Render Holdouts from In Alpha to Separate AOV
2. In your compositor take the alpha of the main image and invert it. 
3. Take the separate hold out AOV and invert that as well
4. Multiple the inverted alpha and inverted holdout AOV
5. Bring in your plate and multiply that with the result from step 4
6. Use an Add node and add together the beauty and the result from 5.

Here's a screenshot of the Blender compositor. It might be a bit hard to see, but there are reflections from the render. 
![Uploading Screenshot 2024-11-26 at 3.54.55 PM.png…]()

**Comment by Corefrecs** [comment]
Thanks for investigating this!
Your suggestion works quiet well and is a good workaround.

I've tried a little bit further and as soon as two area lights are involved, strange things happen with the alpha channel (first image without holdout, second image the alpha channel with holdout):
![no_holdout](https://github.com/user-attachments/assets/502711f7-5b0a-400b-b4c9-ff50852195ac) 
![holdout_alpha](https://github.com/user-attachments/assets/d43249d0-4b58-4a10-99dc-041f88f9ee3c)

Adaptive sampling was deactivated, thought first it had something to do with it.

This pattern is also visible in shadows without holdout and very low samples:
![low_sample](https://github.com/user-attachments/assets/5f5feb86-d5c6-4b4a-afdd-383cd6089e02)

So I'm not sure if it is related to holdouts or they just exaggerate this.

**Comment by Corefrecs** [comment]
Update on this:
the strange Alpha is caused by "Learn Light Results" under Advanced.
Sry if this is known, i'm fairly knew to Renderman.
Unfortunately this doesn't fix the missing reflections.

**Comment by prman-pixar** [comment]
re: learn light results

Yes, it's best to turn this off when you're rendering holdouts. Glad you were able to figure this out

re: reflections

I can't reproduce what you're seeing when I add a second rect light to the scene that you sent before. Looking at the image, it looks like the second light might be really hot, so it's overblowing everything to the point that the reflections disappear?

**Comment by Corefrecs** [comment]
Yes you're right, i boosted the lights and brightened the materials in this scene to better show the noise pattern in the shadows. 
That test was purely for the strange alpha without paying attention to the reflections, i'm sorry for the confusion!
